source/extensions/matching/input_matchers/ip/config.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "envoy/extensions/matching/input_matchers/ip/v3/ip.pb.h"
+#include "envoy/extensions/matching/input_matchers/ip/v3/ip.pb.validate.h"
+#include "envoy/matcher/matcher.h"
+#include "envoy/server/factory_context.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/matching/input_matchers/ip/matcher.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace IP {
+
+class IPConfig : public Envoy::Matcher::InputMatcherFactory {
```

Comment:
nit: name this `Config` to avoid stutter

I also think don't need to qualify `Envoy::` here

Reply:
`IPConfig` renamed in `Config` in https://github.com/envoyproxy/envoy/pull/16592/commits/a0360a73c6b3df1e3a75191f813939fe8ce49608 .

At least Clang 12 isn't happy if I remove `Envoy::` , so I didn't remove it.